### PR TITLE
Add lessons learned from the go-dev-container project

### PR DESCRIPTION
**Pull request by sarg3nt**
- Remove dev container code from .devcontainer
- Update mise.toml to not set versions on golangci-ling and goreleaser
- Add SECURITY.md
- Update chart to pull tag from appVersion
- fix dev.sh
- Add workflows from go-dev-container
- Auto change apVersion in chart for new build
- Add CODEOWNERS

### CODE REVIEW

File: .github/workflows/release-weekly.yml [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+  schedule:
+    # If the schedule stops working, see the docs: https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#schedule
+    - cron: '0 0 * * 0' # Runs weekly at 00:00 on Sunday
+  workflow_dispatch: # Allows manual triggering
+
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}
+  VERSION: ""
+  TAG_MAJOR: ""
```
Comment (aidanleuck) [inline .github/workflows/release-weekly.yml:13]: Do one of the actions modify these empty variables to something not empty?

File: .github/workflows/release-weekly.yml [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+            echo "Could not determine the latest tag version."
+            exit 1
+          fi
+
+      # Docs: https://github.com/marketplace/actions/create-release
+      - name: 'Create Release'
+        id: create_release
+        uses: ncipollo/release-action@2c591bcc8ecdcd2db72b97d6147f871fcd833ba5
+        with:
+          body: "A Weekly release contianing upgrades to system packages in the base Rocky Linux container."
```
Comment (Stanrofl) [inline .github/workflows/release-weekly.yml:82]:
```suggestion
          body: "A Weekly release containing upgrades to system packages in the base Rocky Linux container."
```

**Review by aidanleuck — CHANGES_REQUESTED** [review]
Address question/comment

File: SECURITY.md [hunk 3]
```diff
@@ -0,0 +1,14 @@
+# Security Policy
+
+## Supported Versions
+
+- OS patches are applied Weekly, typically at 00:00 on Sunday, which include applying security patches during the `docker build`, a release is automatically cut adding one to the patch version and a new Docker container is built.
+- A new Helm chart is released when new OS patches are applied, the `appVersion` is auto updated. 
+- The user is responsable for applying the new Helm chart.
```
Comment (Stanrofl) [inline SECURITY.md:7]:
```suggestion
- The user is responsible for applying the new Helm chart.
```

**Review by Stanrofl — CHANGES_REQUESTED** [review]
Fix some minor spelling errors

File: .github/workflows/release-weekly.yml (hunk 1 above)
Comment (sarg3nt) [reply .github/workflows/release-weekly.yml:13]: The weird pipe into GITHUB_ENV sets them.  Yeah, super weird.

Comment (aidanleuck) [reply .github/workflows/release-weekly.yml:13]: That's wild. You can push environment variables up to the pipeline in Github.

Comment (sarg3nt) [reply .github/workflows/release-weekly.yml:13]: Yeah, actually very cool for when an action doesn't do what you want.  Actions have output vars you can user in other actions or even other workflows.

**Pull request merged**